SwrveSDK/src/main/java/com/swrve/sdk/SwrveBase.java

```diff
@@ -94,18 +94,28 @@ public static String getVersion() {
         return null;
     }
 
-    protected T onCreate(final Activity activity) throws IllegalArgumentException {
+    protected void onCreate(final Activity activity) throws IllegalArgumentException {
         if (destroyed) {
             destroyed = initialised = false;
             bindCounter.set(0);
         }
         if (!initialised) {
             // First time it is initialized
-            return init(activity);
+            initInBackground(activity);
```

Comment:
You can not do it just like that. What if user tries to access any Swrve methods before it gets fully initialized in the background?

Reply:
Actually that can happens also without this change. Swrve is initialized with the first activity creation, but an app can be started without user interaction (receiving a push, for example), so Swrve code can be executed also without calling init().

My point is: If that is a problem, it should be prevented outside of the scope of this review.

Reply:
The SDK is meant to be used from user activity, and after the first Activity has been created.

Unfortunately many things will have to move to be async when we move the initialisation to be in a separate thread.

Reply:
> Actually that can happen also without this change. Swrve is initialized with the first activity creation, but an app can be started without user interaction (receiving a push, for example), so Swrve code can be executed also without calling init().
> 
> My point is: If that is a problem, it should be prevented outside of the scope of this review.

Before this change, you could safely assume that calling any Swrve methods in any Activity callback is safe. Now it's not. 
I don't think it's a way to go, even though I'm really waiting for this change

Reply:
In an activity callback, but for example calling a swrve method when a swrve push arrives will be as unsafe as you suggest.

@Sergio-Mira, there is no reference in the documentation saying that

Reply:
Will talk to the team to make it more explicit in the docs